### DISCUSSION TITLE
How to nest scopes?

### DISCUSSION THREAD

**Discussion by cupcakearmy** [actix/actix-web#2539] [Q&A] [answered]
I'm trying some stuff out and can't get nested scopes to work.

What am I missing?

```rust
pub fn init(cfg: &mut web::ServiceConfig) {
  cfg.service(
    web::scope("/api/")
      .service(
        web::scope("/sub/")
          .service(one)
          .service(two)
          .service(three),
      )
      .service(four),
  );
}
```

**Comment by aliemjay** [comment, marked as answer]
You need to remove the trailing slash in scopes. Currently it only matches `/api//sub//other_routes`.

See https://docs.rs/actix-web/4.0.0-beta.15/actix_web/dev/struct.ResourceDef.html#pattern-format-and-matching-behavior .

**Reply by cupcakearmy** [reply]
Thanks! I thought I tried that but apparently not 🤦‍♂️
